src/plugins/core/sheet.ts

```diff
@@ -651,6 +651,9 @@ export class SheetPlugin extends CorePlugin<SheetState> implements SheetState {
     const { orderedSheetIds, sheets } = this;
     const name = cmd.name && cmd.name.trim().toLowerCase();
 
+    if (name && name.length > 31) {
+      return CommandResult.SheetNameTooLong;
```

Comment:
This can break existing spreadsheets.
Let's say the initial revisions look like this:
1. `{ type: "RENAME_SHEET", name: "A very very[...]long name" }`
1. `{ type: "UPDATE_CELL": content: "='A very very[...]long name'!A1" }`

With this commit, the rename would be refused, which would make the cell update invalid.
This would be better for master and it technically requires an upgrade script to "fix" the commands in the database.

Reply:
@LucasLefevre, I have a couple of questions:

1. If a sheet name exceeds the new length limit, what should be the updated name?

2. Do we need to modify all commands that reference sheet names to comply with the new character limit?

Reply:
Hello @dhrp-odoo 

> Do we need to modify all commands that reference sheet names to comply with the new character limit?

Yes, that's what it implies ... and it's very cumbersome... :/

We discussed about it with the team. The real issue comes with excel export. An easy and acceptable solution (at least for stable) is to trigger an warning before exporting if any sheet have a name longer than 31 characters, asking the user to rename the sheet. Nobody ever complained about that limitation. In master, we might do at some point an automatic cropping at the excel export. But it's not a priority though.